### ISSUE TITLE
Catch custom exceptions using @api.errorhandler

### ISSUE THREAD

**Issue by mikelerch** [noirbizarre/flask-restplus#744]
```python
from flask import Flask
from flask_restplus import Resource, Api

app = Flask(__name__)
api = Api(app)


# For example a sqlalchemy.orm.exc.NoResultFound exception
class NoResultFound(Exception):
    pass


@api.errorhandler(NoResultFound)
def handle_no_result_exception(error):
    """Return a custom not found error message and 404 status code"""
    return {'message': 'Not Found'}, 404


@api.route('/hello')
class HelloWorld(Resource):
    def get(self):
        raise NoResultFound


if __name__ == '__main__':
    app.run(debug=True)

```

### **Expected Behavior**
Flask-RESTPlus should handle the custom exception and return the appropriate message and status code.

### **Actual Behavior**
A 500 Internal Server Error is raised, the custom `errorhandler` is ignored.

### **Error Messages/Stack Trace**
```html
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
  "http://www.w3.org/TR/html4/loose.dtd">
<html>
  <head>
    <title>NoResultFound // Werkzeug Debugger</title>

...
```

### **Environment**
- **Python** 3.7
- **Flask** 1.1.1
- **Flask-RESTPlus** 0.13.0

### **Additional Context**

I searched on Google and found a few people having similar issues. From what I understand, only exceptions from `werkzeug.exceptions (HTTPExceptions)` are actually caught.

The [docs](https://flask-restplus.readthedocs.io/en/stable/errors.html#the-api-errorhandler-decorator) are not really explaining this kind of behaviour. How would I solve this issue when using SQLAlchemy for example? 
A possible solution could be catching the `NoResultFound` exception raised by `SQLAlchemy` and raise a new `NotFound` exception from `werkzeug` instead.

Another way could be using the original `errorhandler` provided by Flask. https://flask.palletsprojects.com/en/1.1.x/patterns/apierrors/#implementing-api-exceptions)

Is there a better way of using custom exceptions with Flask-RESTPlus?

**Comment by j5awry** [comment]
This is fixed in PR #693 . We are prepping 0.14.0 for release. in the meantime, if you pull master, errorhandler should work as expected.

If the fix did not address the problem, please re-open.

**Comment by MyJoiT** [comment]
I have the same issue when I use `@app.errorhandler`
